Sources/News.php

```diff
@@ -1760,101 +1893,884 @@ function getXmlProfile($xml_format)
+							),
+						),
+					),
+					array(
+						'tag' => 'link',
+						'attributes' => array('label' => $txt['url']),
+						'content' => $scripturl . '?msg=' . $row['id_msg'],
+					),
+					array(
+						'tag' => 'time',
+						'attributes' => array('label' => $txt['date'], 'UTC' => gmstrftime('%F %T', $row['poster_time'])),
+						'content' => $smcFunc['htmlspecialchars'](strip_tags(timeformat($row['poster_time']))),
+					),
+					array(
+						'tag' => 'modified_time',
+						'attributes' => !empty($row['modified_time']) ? array('label' => $txt['modified_time'], 'UTC' => gmstrftime('%F %T', $row['modified_time'])) : null,
+						'content' => !empty($row['modified_time']) ? $smcFunc['htmlspecialchars'](strip_tags(timeformat($row['modified_time']))) : null,
+					),
+					array(
+						'tag' => 'modified_by',
+						'attributes' => !empty($row['modified_name']) ? array('label' => $txt['modified_by']) : null,
+						'content' => !empty($row['modified_name']) ? $row['modified_name'] : null,
+						'cdata' => true,
+					),
+					array(
+						'tag' => 'modified_reason',
+						'attributes' => !empty($row['modified_reason']) ? array('label' => $txt['reason_for_edit']) : null,
+						'content' => !empty($row['modified_reason']) ? $row['modified_reason'] : null,
+						'cdata' => true,
+					),
+					array(
+						'tag' => 'likes',
+						'attributes' => array('label' => $txt['likes']),
+						'content' => $row['likes'],
+					),
+					array(
+						'tag' => 'approval-status',
+						'attributes' => $show_all ? array('label' => $txt['approval_status']) : null,
+						'content' => $show_all ? $row['approved'] : null,
+					),
+					array(
+						'tag' => 'attachments',
+						'attributes' => array('label' => $txt['attachments']),
+						'content' => $attachments,
+					),
+				),
+			);
+		}
+	}
+	$smcFunc['db_free_result']($request);
+
+	return $data;
+}
+
+/**
+ * Get a user's personal messages.
+ * Only the user can do this, and no one else -- not even the admin!
+ *
+ * @param string $xml_format The XML format. Can be 'atom', 'rdf', 'rss', 'rss2' or 'smf'
+ * @param bool $ascending If true, get the oldest PMs first. Default false.
+ * @return array An array of arrays containing data for the feed. Each array has keys corresponding to the appropriate tags for the specified format.
+ */
+function getXmlPMs($xml_format, $ascending = false)
+{
+	global $scripturl, $modSettings, $board, $txt, $context, $user_info;
+	global $query_this_board, $smcFunc, $sourcedir, $cachedir;
+
+	// Personal messages are supposed to be private
+	if (empty($context['xmlnews_uid']) || ($context['xmlnews_uid'] != $user_info['id']))
+		return array();
+
+	$select_id_members_to = $smcFunc['db_title'] === POSTGRE_TITLE ? "string_agg(pmr.id_member, ',')" : 'GROUP_CONCAT(pmr.id_member)';
+
+	$select_to_names = $smcFunc['db_title'] === POSTGRE_TITLE ? "string_agg(COALESCE(mem.real_name, mem.member_name), ',')" : 'GROUP_CONCAT(COALESCE(mem.real_name, mem.member_name))';
```

Comment:
If someone with a working PostgreSQL setup can test this, that'd be appreciated. It can be tested easily enough by pointing your browser to `?action=.xml;sa=pms;type=smf`

Reply:
"string_agg(pmr.id_member, ',')" -> "string_agg(pmr.id_member::text, ',')"
 than the export is working for pg

Reply:
Thanks!